clients/src/main/java/org/apache/kafka/common/record/FileRecords.java

```diff
@@ -456,6 +456,23 @@ private static FileChannel openChannel(File file,
         }
     }
 
+    @Override
+    public boolean equals(Object o) {
```

Comment:
I wonder if we should include some additional fields. It would be a little odd if slices of different ranges are considered equal. Perhaps we can include `start`, `end` and `isSlice`? Admittedly, this would make less sense from the perspective of `FileChannelRecordBatch` if it relies on the implementation from `FileRecords`. Alternatively,  perhaps we could have `FileChannelRecordBatch` use `fileRecords.channel()` in its `equals` and `hashCode`?

Sigh, if only `FileChannel` gave us a more useful `toString` 😞

Reply:
I had a go with the second choice, it does not look super good, but it should get the job done 😄


Alternately, `FileChannelRecordBatch` could expose a `protected FileChannel channel()` method.

The equals bit would look something like:
```java
return ...
       ...
       (channel() == null ? that.channel() == null : channel().equals(that.channel());
```

No strong opinion between these options